core/node/state_keeper/src/keeper.rs

```diff
@@ -104,9 +105,14 @@ impl ZkSyncStateKeeper {
     }
 
     /// Fallible version of `run` routine that allows to easily exit upon cancellation.
-    async fn run_inner(&mut self) -> Result<Infallible, Error> {
+    async fn run_inner(
+        &mut self,
+        mut stop_receiver: watch::Receiver<bool>,
+    ) -> Result<Infallible, Error> {
         let (cursor, pending_batch_params) = self.io.initialize().await?;
         self.output_handler.initialize(&cursor).await?;
+        self.health_updater
+            .update(Health::from(HealthStatus::Ready));
```

Comment:
- Why not update health from `cursor` immediately, like you do below?
- To be really nitpicky, one could argue that the state keeper is affected until it processes the pending batch, since it doesn't respond to environment signals (mempool txs and timeouts) until then. This is subjective though, so feel free to ignore.

Reply:
Addressed in https://github.com/matter-labs/zksync-era/pull/3193/commits/d2c983c9e3685e7958bdf4a40c82ada48090889c